### ISSUE TITLE
Override Continue button handler in Informed Consent.

### ISSUE THREAD

**Issue by penacristian** [ResearchKit/ResearchKit#996]
I am trying to track the informed consent progress so the next time the app is opened I can start from where I left over. For that purpose, I want to override the continue button handler.

How can I do that? Is there a better way?

Thanks

**Comment by Pariecemckinney-apple** [comment]
@penacristian, When you say the continue button are you referencing the "next" button. If so, are you trying to edit the handler for one button at a certain point or for every "next" button that the user clicks? Either way there are delegate methods such as `func taskViewController(_ taskViewController: ORKTaskViewController, didChange result: ORKTaskResult)` that give you a snapshot of the current results object so that you can do whatever it that you want with it. Also, there is `func taskViewController(_ taskViewController: ORKTaskViewController, viewControllerFor step: ORKStep)` that will be invoked when a step start and ends.

Hope this helps, and please feel free to reach back out if you are still having issues.

**Comment by penacristian** [comment]
Yes. I have that method. The problem that I'm facing is that I have a visual step created from a consent document with consent sections, which turns into multiple screens with multiple "next" buttons, but that delegate method only gets called with the top "layer" of steps and not with the single sections.

**Comment by Pariecemckinney-apple** [comment]
@penacristian - Thank you for explaining. As of now RK does not provide any delegate methods that notify you when a section has changed. This is a interesting feature that I we will definitely begin investigating. 

With that being said, the functionality that you're looking for would have to be added by the developer currently. Feel free to reach back out if there is anything else you need.

**Comment by Pariecemckinney-apple** [comment]
closing due to inactivity
